src/executor/init/MigrationInitExecutor.ts

```diff
@@ -1,14 +1,22 @@
 import ElasticsearchClient from '../../utils/es/ElasticsearchClient';
-import { MAPPING_HISTORY_INDEX_NAME, SearchEngineVersion } from '../../model/types';
+import {
+    MappingHistoryIndex,
+    MAPPING_HISTORY_INDEX_NAME,
+    SearchEngineVersion
+} from '../../model/types';
 import { cli } from 'cli-ux';
 import v7Mapping from '../../resources/mapping/migrate_history_esV7.json';
 import v6Mapping from '../../resources/mapping/migrate_history_esV6.json';
 
 export async function createHistoryIndex(
     esClient: ElasticsearchClient,
-    esVersion?: SearchEngineVersion
+    esVersion?: SearchEngineVersion,
+    numShards?: number,
+    numReplica?: number
 ): Promise<void> {
-    const mappingData = mapping(esVersion);
+    const mappingData: MappingHistoryIndex = mapping(esVersion);
+    if (typeof numShards !== 'undefined') mappingData.settings.index.number_of_shards;
+    if (typeof numReplica !== 'undefined') mappingData.settings.index.number_of_replicas;
```

Comment:
Expression statement is not assignment or call .
Please check if it's working as expected.

Reply:
for reference, here are my local test results:

**without s&r options)**
```
$ ./bin/run init
Start creating index for migrate.
Finish creating index for migrate.

$ curl localhost:9202/_cat/indices?v=true
health status index           uuid                   pri rep docs.count docs.deleted store.size pri.store.size
yellow open   migrate_history J_zQ3S9_QjGcdF7AmecaDA   1   2          0            0       208b           208b
```

**with s&r options)**
```
$ ./bin/run init -s 1 -r 0                     
Start creating index for migrate.
Finish creating index for migrate.

$ curl localhost:9202/_cat/indices?v=true      
health status index           uuid                   pri rep docs.count docs.deleted store.size pri.store.size
green  open   migrate_history xaIv1icRQ5ieXpmzbC6g1Q   1   0          0            0       208b           208b
```

Reply:
Thanks